sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -6,15 +6,506 @@
+
+    def get_vendor_OUI(self):
+        return self.xcvr_eeprom.read(consts.VENDOR_OUI_FIELD)
+
+    def get_vendor_date(self):
+        return self.xcvr_eeprom.read(consts.VENDOR_DATE)
+
+    def get_connector_type(self):
+        return self.xcvr_eeprom.read(consts.CONNECTOR_TYPE)
+
+    def get_module_media_type(self):
+        return self.xcvr_eeprom.read(consts.MODULE_MEDIA_TYPE)
+
+    def get_host_electrical_interface(self):
+        return self.xcvr_eeprom.read(consts.HOST_ELECTRICAL_INTERFACE)
+
+    def get_module_media_interface(self):
+        media_type = self.get_module_media_type()
+        if media_type == 'Multimode Fiber (MMF)':
+            return self.xcvr_eeprom.read(consts.MODULE_MEDIA_INTERFACE_850NM)
+        elif media_type == 'Single Mode Fiber (SMF)':
+            return self.xcvr_eeprom.read(consts.MODULE_MEDIA_INTERFACE_SM)
+        elif media_type == 'Passive Copper Cable':
+            return self.xcvr_eeprom.read(consts.MODULE_MEDIA_INTERFACE_PASSIVE_COPPER)
+        elif media_type == 'Active Cable Assembly':
+            return self.xcvr_eeprom.read(consts.MODULE_MEDIA_INTERFACE_ACTIVE_CABLE)
+        elif media_type == 'BASE-T':
+            return self.xcvr_eeprom.read(consts.MODULE_MEDIA_INTERFACE_BASE_T)
+        else: 
+            return 'Unknown media interface'
+    
+    def get_host_lane_count(self):
+        lane_count = self.xcvr_eeprom.read(consts.LANE_COUNT)
+        return (lane_count >> 4) & 0xf
+
+    def get_media_lane_count(self):
+        lane_count = self.xcvr_eeprom.read(consts.LANE_COUNT)
+        return (lane_count) & 0xf
+
+    def get_host_lane_assignment_option(self):
+        return self.xcvr_eeprom.read(consts.HOST_LANE_ASSIGNMENT_OPTION)
+
+    def get_media_lane_assignment_option(self):
+        return self.xcvr_eeprom.read(consts.MEDIA_LANE_ASSIGNMENT_OPTION)
+
+    def get_active_apsel_hostlane1(self):
+        result = self.xcvr_eeprom.read(consts.ACTIVE_APSEL_HOSTLANE_1)
+        return (result >> 4) & 0xf
+
+    def get_active_apsel_hostlane2(self):
+        result = self.xcvr_eeprom.read(consts.ACTIVE_APSEL_HOSTLANE_2)
+        return (result >> 4) & 0xf
+
+    def get_active_apsel_hostlane3(self):
+        result = self.xcvr_eeprom.read(consts.ACTIVE_APSEL_HOSTLANE_3)
+        return (result >> 4) & 0xf
+
+    def get_active_apsel_hostlane4(self):
+        result = self.xcvr_eeprom.read(consts.ACTIVE_APSEL_HOSTLANE_4)
+        return (result >> 4) & 0xf
+
+    def get_active_apsel_hostlane5(self):
+        result = self.xcvr_eeprom.read(consts.ACTIVE_APSEL_HOSTLANE_5)
+        return (result >> 4) & 0xf
+
+    def get_active_apsel_hostlane6(self):
+        result = self.xcvr_eeprom.read(consts.ACTIVE_APSEL_HOSTLANE_6)
+        return (result >> 4) & 0xf
+
+    def get_active_apsel_hostlane7(self):
+        result = self.xcvr_eeprom.read(consts.ACTIVE_APSEL_HOSTLANE_7)
+        return (result >> 4) & 0xf
+
+    def get_active_apsel_hostlane8(self):
```

Comment:
All these APIs can be combined to one API that returns a dictionary{"Lane no" : Apsel code}

Reply:
Done